sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PortEvaluationStage.java

```diff
@@ -0,0 +1,125 @@
+package com.mesosphere.sdk.offer.evaluate;
+
+import com.mesosphere.sdk.offer.*;
+import org.apache.mesos.Protos;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.IntStream;
+
+/**
+ * This class evaluates an offer for its port resources against an {@link OfferRequirement}, finding ports dynamically
+ * in the offer where specified by the framework and modifying {@link org.apache.mesos.Protos.TaskInfo} and
+ * {@link org.apache.mesos.Protos.ExecutorInfo} where appropriate so that the ports are available in their respective
+ * environments.
+ */
+public class PortEvaluationStage extends ResourceEvaluationStage implements OfferEvaluationStage {
+    private String portName;
+    private Integer port;
+
+    public PortEvaluationStage(Protos.Resource resource, String taskName, String portName, Integer port) {
```

Comment:
Should `port` ever be null? If so, then add a null check on assignedPort at L42. If not, then use an `int` instead.